network/src/primary.rs

```diff
@@ -72,21 +71,25 @@ impl PrimaryNetwork {
         &mut self,
         address: Multiaddr,
         message: BincodeEncodedPayload,
-    ) -> CancelHandler<()> {
+    ) -> CancelHandler<MessageResult> {
         let client = self.client(address);
+
+        let message_send = move || {
+            let mut client = client.clone();
+            let message = message.clone();
```

Comment:
surely this is not deep-copying the entire payload every time... right?

Reply:
As of today the message type is essentially a Bytes struct which has a very efficient clone operation (its essentially just an atomic increment)

Reply:
The PR does not change the cloning behavior, but FWIW the client also has a cheap cloning operation.